### PR TITLE
fix: execute after hook.

### PR DESCRIPTION
**Pull request by appleboy**
Fix some bugs from https://github.com/go-gitea/gitea/issues/677#issuecomment-273945141

cc @lunny

### CODE REVIEW

File: session_delete.go [hunk 1]
```diff
@@ -226,7 +226,8 @@ func (session *Session) Delete(bean interface{}) (int64, error) {
 				session.afterDeleteBeans[bean] = &afterClosures
 			}
 		} else {
-			if _, ok := interface{}(bean).(AfterInsertProcessor); ok {
+			if processor, ok := interface{}(bean).(AfterInsertProcessor); ok {
```
Comment (lunny) [inline session_delete.go]: AfterInsertProcessor ?

Comment (appleboy) [inline session_delete.go]: Done

**Comment by appleboy** [comment]
@lunny Please review again.

**Comment by lunny** [comment]
Could you help add some tests PR int github.com/go-xorm/tests?

**Comment by appleboy** [comment]
@lunny See the PR https://github.com/go-xorm/tests/pull/16

**Comment by lunny** [comment]
@appleboy could you rebase this PR?

**Pull request merged**